Sources/Workspace/Workspace.swift

```diff
@@ -602,11 +601,13 @@ public class Workspace {
             }
         }
         // Commit the pin.
-        try pinsStore.pin(
+        pinsStore.pin(
             package: package,
             repository: dependency.repository,
             state: checkoutState,
             reason: reason)
+        
+        try pinsStore.saveState()
```

Comment:
Instead of adding this here, this can be moved higher in the call hierarchy so this method becomes non-throwing. Then methods such as pinAll will become non throwing. In long term we will want to make the pin method at L489 also non throwing but thats a future thing that needs to be done.